### DISCUSSION TITLE
Question About the context of Functor and Monad

### DISCUSSION THREAD

**Discussion by adispring** [ramda/ramda#3237] [Q&A]
`R.map` will only change the value of functor, but not the structure(context) of the functor, what's the benefit of this in practical programming?

`R.chain` can change both the value and context of monad, I know the benefit of this feature: this can help people process complex thing in a concise way, such as Maybe, Either and  Promise.

But I can not understand what's the benefit for `R.map` keep the context of functor unchanged.

Here is an article: [What Does fmap Preserve?](https://duplode.github.io/posts/what-does-fmap-preserve.html)

**Comment by CrossEye** [comment]
I'm not sure I understand your question.  It seems to me that the fact that `map (square)` does all these things is pretty powerful:

```js
[1, 2, 3, 4] ==> 
[1, 4, 9, 16]

Maybe (5) ==> 
Maybe (25)

(n -> n + 3) ==> 
(n -> (n + 3) * (n + 3))

{v: 8 , c: [{v: 6,  c: [{v: 7},  {v: 5}]},  {v: 3, c: [{v: 0}, {v: 9}]}]} ==> 
{v: 64, c: [{v: 36, c: [{v: 49}, {v: 25}]}, {v: 9, c: [{v: 0}, {v: 81}]}]}
```

In a sense that's very hard to make precise, Functors are a means to convert transformations of values into transformations on structures around those values that maintain the structure around the transformed values.  That seem pretty powerful.  Clearly Monads are more powerful, but there are plenty of things for which we can create Functor instances but not Monad ones.
